frc46_token/src/token/state.rs

```diff
@@ -270,7 +270,14 @@ impl TokenState {
     /// This involves iterating through the entire HAMT
     pub fn count_balances<BS: Blockstore>(&self, bs: &BS) -> Result<usize> {
         let balance_map = self.get_balance_map(bs)?;
-        Ok(balance_map.for_each(|_, _| Ok(())).into_iter().count())
```

Comment:
Was this a bug?

Reply:
Yes - `for_each` returns a `Result`, so it was converting that into an iterator and counting it, which always returned 1 regardless of the underlying tree size.